var/spack/repos/builtin/packages/adios2/package.py

```diff
@@ -208,3 +211,10 @@ def cmake_args(self):
                         % spec['python'].command.path)
 
         return args
+
+    def setup_run_environment(self, env):
+        # Make sure the install lib prefix is in the plugin path so the
+        # HDF5 VOL plugin can be found
+        if '+hdf5' in self.spec and self.spec['hdf5'].satisfies('@1.12:'):
+            os.environ['HDF5_PLUGIN_PATH'] += os.pathsep + \
+                os.path.join(self.spec.prefix, 'lib')
```

Comment:
```suggestion
            env.append_path('HDF5_PLUGIN_PATH', self.prefix.lib)
```
It would be even better to use `self.spec['adios2'].libs.directories[0]` in case the libraries can be installed in `lib` or `lib64`.

Reply:
`append_path` is certainly useful, I'll make the change accordingly.  wrt the lib directory, CMake tries to introspect the system and might use `lib`, might use `lib64` depending on the distro.  That's why I explicitly set the install dir to `lib` up top in this pr to avoid the ambiguity.  Does using `self.prefix.lib` just get automatically populated by spack to be the correct location?  if so then yueah, I can just use that and not worry about the `-DCMAKE_INSTALL_LIBDIR` addition.

Reply:
`self.prefix.lib` always uses `lib`, `self.spec['adios2'].libs.directories[0]` checks both `lib` and `lib64` and returns the one where the libraries were found.

Reply:
Is there a spack convention for what the directory layout should be if possible?

Reply:
Would `self.spec['adios2'].libs.directories[0]` be any different that just `self.spec.libs.directories[0]`?

Reply:
Conventions aren't necessarily helpful. Spack supports external packages, where we can't control whether they use lib or lib64. For that reason, I try to make sure both will work. There is a PR that adds `CMAKE_INSTALL_LIBDIR` for all CMake packages, but I don't think it's been merged yet, and it won't solve the external package problem.

`spec.libs` and `spec['adios2'].libs` are slightly different. I forget the details, but @alalazo was explaining to me once why the latter works and the former doesn't.

Reply:
Good enough for me.

Reply:
I took a different approach and instead just locate the actual library and use it's path.